Shared/StringToDatum.h

```diff
@@ -0,0 +1,268 @@
+#ifndef STRING_TO_DATUM_H
+#define STRING_TO_DATUM_H
+
+#include <Utils/StringConversions.h>
+#include <boost/utility/string_view.hpp>
+#include <cinttypes>
+#include "../Shared/StringToDatum.h"
+#include "DateConverters.h"
+#include "TimeGM.h"
+#include "sqltypes.h"
+
+template <typename String>
```

Comment:
Would it be better if we keep the implementations in cpp file?
To avoid compile errors, one can include cpp file at the end of the header file.

Reply:
You mean creating StringToDatum.cpp and include it at the end of StringToDatum.h?

Reply:
Yes